packages/dds/tree/src/test/shared-tree/undo.spec.ts

```diff
@@ -673,6 +675,60 @@ describe("Undo and redo", () => {
 			"Error: Unable to revert a revertible that has been disposed.",
 		);
 	});
+
+	it("clone list of revertibles", () => {
+		const view = createInitializedView();
+		const { undoStack } = createTestUndoRedoStacks(view.events);
+
+		assert(view.root.child !== undefined);
+		view.root.child.propertyOne = 256; // 128 -> 256
+		view.root.child.propertyTwo.itemOne = "newItem"; // "" -> "newItem"
+
+		const forkedView = view.fork();
+
+		const batchedRevertibles: RevertibleAlpha[] = [];
+		for (const revertible of undoStack) {
+			batchedRevertibles.push(revertible);
+		}
+
+		const clonedRevertibles = cloneRevertiblesInBatch(batchedRevertibles, forkedView);
+
+		assert.equal(clonedRevertibles.length, 2);
+		assert.equal(forkedView.root.child?.propertyOne, 256);
+		assert.equal(forkedView.root.child?.propertyTwo.itemOne, "newItem");
+
+		assert.equal(clonedRevertibles[0]?.status, RevertibleStatus.Valid);
+		assert.equal(clonedRevertibles[1]?.status, RevertibleStatus.Valid);
+
+		clonedRevertibles.pop()?.revert();
+		assert.equal(forkedView.root.child?.propertyOne, 256);
+		assert.equal(forkedView.root.child?.propertyTwo.itemOne, "");
+
+		clonedRevertibles.pop()?.revert();
+		assert.equal(forkedView.root.child?.propertyOne, 128);
+	});
+
+	it("cloning list of disposed revertibles throws error", () => {
```

Comment:
Can you add another test for passing in a view that doesn't have the same changes?